src/cogent3/core/annotation_db.py

```diff
@@ -37,13 +39,33 @@
 # Define custom types for storage in sqlite
 # https://stackoverflow.com/questions/18621513/python-insert-numpy-array-into-sqlite3-database
 def array_to_sqlite(data):
-    return data.tobytes()
+    out = io.BytesIO()
+    numpy.save(out, data)
+    out.seek(0)
+    return out.read()
 
 
 def sqlite_to_array(data):
-    result = numpy.frombuffer(data, dtype=int)
-    dim = result.shape[0] // 2
-    return result.reshape((dim, 2))
+    out = io.BytesIO(data)
+    out.seek(0)
+    try:
+        result = numpy.load(out)
+    except ValueError:
+        # array is not stored in the numpy.save format
+        # attempt to read from the old format where the
+        # array was saved using numpy.ndarray.tobytes
+        result = numpy.frombuffer(data, dtype=int)
+        dim = result.shape[0] // 2
+        result = result.reshape((dim, 2))
+
+        warnings.warn(
```

Comment:
I've been scratching my head for the last little bit on this. When this code is run through a WIP test I have going on in pytest, pytest is able to capture and display the warning. However, if I take the equivalent code in pytest and run it in its own file, the warnings are not displayed - unless the warning is also specifically given the parameter stacklevel=4.